### DISCUSSION TITLE
Using flux_bootstrap_git with Existing Manifests in Github Repository

### DISCUSSION THREAD

**Discussion by jranai** [fluxcd/flux2#5096] [Q&A] [answered]
Hi all ,

I want to deploy FluxCD and sync existing Kubernetes manifests already stored in my Git repository at a specific folder. 
Here’s my setup:
My Git repo contains the required manifests to deploy FluxCD and other resources. These are already committed under a specific folder. All of my clusters uses this folder as source of truth to deploy Fluxcd
In this usecase I’m trying to use Terraform’s flux_bootstrap_git resource to bootstrap FluxCD.
```
resource "flux_bootstrap_git" "this" {
  count = var.deploy_fluxcd ? 1 : 0

  depends_on             = [module.kubernetes]
  path                   = "apps/preprod/flux-system"
  kustomization_override = <<EOT
resources:
  - gotk-components.yaml
  - gotk-sync.yaml
patches:
  - patch: |
 # some specific patch 
EOT
}
```

My issues so far 
	1.	flux_bootstrap_git creates a new flux-system subdirectory inside existing folder (e.g. apps/preprod/flux-system)  that already contains the manifests, which I don’t want.
	2.	Specifying a broader path (e.g., apps/preprod/) can overwrite or replace existing files in the repository.
	3.	I want Flux to only sync the existing manifests without pushing any new ones.
	
first of all , is this even possible to do using flux_bootstrap_git to begin with
And Second is using one single folder that contains my fluxcd manifests as source truth for all my k8s clusters a good practice  in the first place ? 
	
Thanks in advance for your help!

**Comment by stefanprodan** [comment, marked as answer]
The `flux bootstrap` command and its Terraform counterpart `flux_bootstrap_git` are designed to configure each cluster with its own Flux instance manifests. Reusing the Flux manifests across clusters is not supported, the idea here is that you would roll Flux upgrades gradually without risking to bring down all the clusters due to some misconfiguration. 

An alternative solution to bootstrap, where you don't push to Git from Terraform, is by using [Flux Operator](https://github.com/controlplaneio-fluxcd/flux-operator). See here an example https://github.com/controlplaneio-fluxcd/flux-operator/tree/main/config/terraform

**Comment by XtremeAI** [comment]
Hey guys,  @stefanprodan 

Got the same question as I tried to swap my helm based Flux install in Terraform to Flux provider just yesterday but failed. I've got all manifests including gotk-components.yaml & gotk-sync.yaml of the version and adjustments I want already available on 'path' in my git repo. I just want Terraform to install Flux controllers that would reconcile with the existing manifests.

I've got 1 gotk-components.yaml & gotk-sync.yaml file per version that is used by 40+ clusters with no problem so far and I assure myself that no misconfiguration is done and/or no more than a test cluster is affected.

So it would be good to add support for this option.

Thanks!

**Reply by stefanprodan** [reply]
You may want to give [Flux Operator](https://github.com/controlplaneio-fluxcd/flux-operator) a try which doesn't need the `gotk-components.yaml` in Git anymore so you can easly scale to lots of clusters. Terraform example here: https://github.com/controlplaneio-fluxcd/flux-operator/tree/main/config/terraform
